pkg/storage/node.go

```diff
@@ -96,6 +99,17 @@ func (s *nodeStorage) Store(batch *MetricsBatch) {
 			}
 		}
 	}
+	newTimeStamp := time.Now()
```

Comment:
This code is pretty inefficient. Every time we scrape a node, we scan all nodes if their metrics are outdated. This gives us O(n^2) time and memory complexity to scrape and store metrics from n nodes. 

If this was not picked up in the tests you prepared, means that they were not accurate. Please look into writing a benchmark for this.

Reply:
OK. I will optimize it